tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -1754,6 +1754,46 @@ public function test_get_items_all_post_formats() {
 		$this->assertSame( $formats, $data['schema']['properties']['format']['enum'] );
 	}
 
+	/**
+	 * @ticket 55593
+	 */
+	public function test_get_items_parent_ids_primed() {
+		$parent_id1 = self::$post_ids[0];
+		$parent_id2 = self::$post_ids[1];
+		$parent_ids = array( $parent_id2, $parent_id1 );
+
+		$this->factory->attachment->create_object(
+			DIR_TESTDATA . '/images/canola.jpg',
+			$parent_id1,
+			array(
+				'post_mime_type' => 'image/jpeg',
+				'post_excerpt'   => 'A sample caption 1',
+			)
+		);
+
+		$this->factory->attachment->create_object(
+			DIR_TESTDATA . '/images/canola.jpg',
+			$parent_id2,
+			array(
+				'post_mime_type' => 'image/jpeg',
+				'post_excerpt'   => 'A sample caption 2',
+			)
+		);
+
+		// Attachment creation warms parent ids. Needs clean up for test.
+		wp_cache_delete_multiple( $parent_ids, 'posts' );
+
+		$filter = new MockAction();
+		add_filter( 'update_post_metadata_cache', array( $filter, 'filter' ), 10, 2 );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/media' );
+		rest_get_server()->dispatch( $request );
+
+		$args = $filter->get_args();
+		$last = end( $args );
+		$this->assertEqualSets( $parent_ids, $last[1] );
```

Comment:
Do we want to verify that in fact `last` is an array before asserting this line?

```suggestion
		$this->assertIsArray( $last );
		$this->assertEqualSets( $parent_ids, $last[1] );
```

Reply:
If last is not an array, the `assertEqualSets` will fail and test will fail. What is the benefit of this check?

Reply:
```suggestion
                 $this->assertIsArray( $last, 'The last value is not an array' );
		$this->assertEqualSets( $parent_ids, $last[1] );
```

Reply:
The main benefit is the error coming out of the assertion as you would get a clear error message indicating that the assertion failed due `last` is not an array and the opposite the test will fail not because of the assertion but because of accessing an invalid index on not array value in this case `null[1]`